### PR TITLE
[Program Migration] Save imported programs

### LINKED ISSUE #7525: [Program Migration] Import Program Definitions
Add the ability for program admins to submit the json generated by exporting a program and for that json to be translated into a program definition that is saved in the db.

Part of #7087

### PR DESCRIPTION
**Pull request by rockycodes**
### Description

This PR adds the functionality to import programs into a new environment. Importing questions will be added in a follow-up PR: https://github.com/civiform/civiform/issues/7527

The main changes are passing the `json` along to `AdminImportViewPartial` and adding a `saveProgram` method and route.

#### General

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Instructions for manual testing

1. Enable the `PROGRAM_MIGRATION_ENABLED` flag
2. Add a program (can seed programs and questions via the developer view)
3. Export the program via the export tab and copy the exported json
4. Paste the json into the import tab and change the admin id of the program to avoid collisions (and optionally the program name so you can tell the programs apart)
5. Click "Display program information" and confirm that the correct info is displayed
6. Click "Save Program"
7. Navigate to the "Programs" tab and see your imported program there 

### Issue(s) this completes

Fixes #7525

### CODE REVIEW

File: server/app/views/admin/migration/AdminImportViewPartial.java [hunk 1]
```diff
@@ -40,21 +47,39 @@ public DomContent renderError(String errorMessage) {
...
+  public DomContent renderProgramData(
+      Http.Request request, ProgramMigrationWrapper programMigrationWrapper, String json) {
     ProgramDefinition program = programMigrationWrapper.getProgram();
     DivTag programDiv =
         div()
             .withId(PROGRAM_DATA_ID)
             .with(
                 h3("Program name: " + program.localizedName().getDefault()),
-                p("Admin name: " + program.adminName()));
+                h4("Admin name: " + program.adminName()));
```
Comment (rockycodes) [inline server/app/views/admin/migration/AdminImportViewPartial.java:58]: The test locator couldn't find this when it was a ptag so I changed it to an `h4`. Maybe related to this bug: https://github.com/microsoft/playwright/issues/23377

File: server/conf/routes [hunk 2]
```diff
@@ -133,11 +133,12 @@ POST    /admin/apiKeys                              controllers.admin.AdminApiKe
...
-POST /admin/export/hx/program    controllers.admin.AdminExportController.hxExportProgram(request: Request)
-POST /admin/export/download/:adminName  controllers.admin.AdminExportController.downloadJSON(request: Request, adminName: String)
-GET /admin/import            controllers.admin.AdminImportController.index(request: Request)
-POST /admin/import/hx/program   controllers.admin.AdminImportController.hxImportProgram(request: Request)
+GET /admin/export                                   controllers.admin.AdminExportController.index(request: Request)
+POST /admin/export/hx/program                       controllers.admin.AdminExportController.hxExportProgram(request: Request)
+POST /admin/export/download/:adminName              controllers.admin.AdminExportController.downloadJSON(request: Request, adminName: String)
+GET /admin/import                                   controllers.admin.AdminImportController.index(request: Request)
+POST /admin/import/hx/program                       controllers.admin.AdminImportController.hxImportProgram(request: Request)
+POST /admin/import/saveProgram                      controllers.admin.AdminImportController.saveProgram(request: Request)
```
Comment (rockycodes) [inline server/conf/routes:141]: `saveProgram` is the only new route. The rest I just cleaned up the tabbing.

File: server/app/controllers/admin/AdminImportController.java [hunk 3]
```diff
@@ -74,7 +79,7 @@ public Result hxImportProgram(Http.Request request) {
         formFactory
             .form(AdminProgramImportForm.class)
             .bindFromRequest(request, AdminProgramImportForm.FIELD_NAMES.toArray(new String[0]));
-    String jsonString = form.get().getProgramJson();
+    String jsonString = form.get().getProgramJSON();
```
Comment (avaleske) [inline server/app/controllers/admin/AdminImportController.java:82]: **nit**: I feel bad about nitting this but the styleguide says it should be `getProgramJson()`:
https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

**Review by avaleske — APPROVED** [review]
LGTM!

File: server/app/controllers/admin/AdminImportController.java (hunk 3 above)
Comment (rockycodes) [reply server/app/controllers/admin/AdminImportController.java:82]: Thank you for the nit! I should have checked the style guide, I wasn't sure.

Comment (rockycodes) [reply server/app/controllers/admin/AdminImportController.java]: Updated.

**Pull request merged**